spec/unit/validator_spec.rb

```diff
@@ -73,5 +72,55 @@ def self.name
       expect(validator).to_not be_valid
       expect(validator.errors[:name]).to eql(['TAKEN!'])
     end
+
+    context 'with unique attributes within a scope' do
+      let(:user_validator) do
+        Class.new {
+          include ROM::Model::Validator
+
+          relation :users
+
+          validates :email, uniqueness: {scope: :name}
+
+          def self.name
+            'User'
+          end
+        }
+      end
+
+      let(:doubly_scoped_validator) do
+        Class.new {
+          include ROM::Model::Validator
+
+          relation :users
+
+          validates :email, uniqueness: {scope: [:name, :birthday]}
+
+          def self.name
+            'User'
+          end
+        }
+      end
+
+      it 'does not add errors' do
+        rom.relations.users.insert(name: 'Jane', email: 'jane+doe@doe.org')
+        attributes = user_attrs.new(name: 'Jane', email: 'jane@doe.org', birthday: Date.parse('2014-12-12'))
+        validator = user_validator.new(attributes)
+        expect(validator).to be_valid
+      end
+
+      it 'adds an error when the doubly scoped validation fails' do
+        attributes = user_attrs.new(name: 'Jane', email: 'jane@doe.org', birthday: Date.parse('2014-12-12'))
+        validator = doubly_scoped_validator.new(attributes)
+        expect(validator).to be_valid
+
+        rom.relations.users.insert(attributes.attributes)
+        expect(validator).to_not be_valid
+
+        attributes = user_attrs.new(name: 'Jane', email: 'jane+doe@doe.org', birthday: Date.parse('2014-12-12'))
+        validator = doubly_scoped_validator.new(attributes)
+        expect(validator).to be_valid
+      end
+    end
```

Comment:
Your concern is definitely valid if the test data were any more generic. However the `doubly_scoped_validator` only does uniqueness validation and there's no other way for validation to fail. It is also named very specifically so that future developers know that it has a singular purpose and it's also [only scoped](https://github.com/vrish88/rom-rails/blob/master/spec/unit/validator_spec.rb#L76-L124) to the describe block that deals with scoped uniqueness validation.

Also in this implementation the test only knows enough about the validator to ensure the behavior is there. This allows things like the error text or `#errors` to change independently of this test.